src/Autofac/Core/Registration/ServiceRegistrationInfo.cs

```diff
@@ -88,10 +88,16 @@ public IEnumerable<IComponentRegistration> Implementations
             get
             {
                 RequiresInitialization();
-                return Enumerable
-                    .Reverse(_defaultImplementations)
-                    .Concat(_sourceImplementations)
-                    .Concat(_preserveDefaultImplementations);
+
+                var resultingCollection = Enumerable.Reverse(_defaultImplementations);
```

Comment:
Everything in this file looks good.

Reply:
I brought all the changes from this file into the `develop` branch.